### PR TITLE
Crash Fixes

### PR DESCRIPTION
**Pull request by Matthew-McRaven**
Fixes a crash where CPUGraphicItems are deleted multiple times on close.
Fixes a crash where changing values in the main memory window would cause a stack overflow.

### CODE REVIEW

File: mainmemory.cpp [hunk 1]
```diff
@@ -241,7 +241,7 @@ void MainMemory::cellDataChanged(QTableWidgetItem *item)
 {
     // disconnect this signal so that modifying the text of the column next to it doesn't fire this signal; reconnect at the end
     disconnect(ui->tableWidget, SIGNAL(itemChanged(QTableWidgetItem*)), this, SLOT(cellDataChanged(QTableWidgetItem*)));
-
+    disconnect(dataSection,&CPUDataSection::memoryChanged,this,&MainMemory::onMemoryValueChanged);
```
Comment (EmilyDimpfl) [inline mainmemory.cpp]: I'm pretty sure that this should instead be:
`disconnect(dataSection, SIGNAL(memoryChanged(quint16,quint8,quint8)), this, SLOT(onMemoryValueChanged(quint16,quint8,quint8)));`
...as that uses the macros supplied by Qt.

Line 272 similar.

**Review by EmilyDimpfl — DISMISSED** [review]
There's some Qt formatting stuff that should probably be changed - I'm not sure if you've used this style elsewhere, but we should be consistent.

**Comment by Matthew-McRaven** [comment]
The syntax used was added in QT5.0

Qt 5.0 adds two new ways to connect signals and slots. One approach uses functors (the one used in my code) and the other uses lambdas. See [QT 5 New Signal Slot Syntax](https://wiki.qt.io/New_Signal_Slot_Syntax)

The syntax on the line above is the string-based signal-slot connection, whereas the form I used is functor-based signal-slot connections.

The functor based signal-slot connection provides compile-time type checking and checks for the existence of slots at compile time instead of run-time. 

Across the project, it would be best to use the new syntax for additional type safety and more useful debugging information when slots argument numbers and types don't match. I have been trying to replace the old syntax as I make upgrades throughout the project.

See [Signal Slot Syntax](http://doc.qt.io/qt-5/signalsandslots-syntaxes.html)

**Comment by EmilyDimpfl** [comment]
Hmm, I think that we should maintain consistency, so that might mean changing formats to Qt5's new syntax. @StanWarford, do you have a preference?

I personally prefer the old format because its more verbose and obvious what signals are being connected to which slots. That said, I don't think we're using overloads so we shouldn't have the "bad" case where we have to do a bunch of casts (a la https://wiki.qt.io/New_Signal_Slot_Syntax#Overload) so we could commit entirely to one style or the other. It looks like we have 173 connect() usages across the entire app, so not necessarily quick and easy to switch to the new style, but not insurmountable either.

I also looked at some of the previous commits, and noticed a few formatting consistency changes, like spaces between arguments. For instance, in the dark mode code:

`connect(this,&MainWindow::darkModeChanged,cpuPaneTwoByteDataBus,&CpuPane::onDarkModeChanged);`
to 
`connect(this, &MainWindow::darkModeChanged, cpuPaneTwoByteDataBus, &CpuPane::onDarkModeChanged);`

**Comment by Matthew-McRaven** [comment]
I switched my local branch to use the new qt5 syntax. It did uncover some type mismatches in our signals and slots. It’s the new recommendation, and the amount of code change is minimal. I will test it some more, then commit it.

**Comment by StanWarford** [comment]
I just had a look at the New Signal Slot Syntax documentation. While it does not mention that the old style will be deprecated in the future, my general philosophy is to adopt the new when convenient to do so. You never know if/when it will be. Might as well be up to date. Seems the tradeoff to have type checking with new is also good, especially since we are not overloading function names as Emily noted. New style would be my vote.

**Pull request merged**